### PR TITLE
fix: Button loading={delay: 0} 不显示加载

### PR DESCRIPTION
**Pull request by YDFlame13**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/45273

### 💡 Background and solution

在`getLoadingConfig`中判断`delay = 0`时，直接设置`loading = true`

### 📝 Changelog

| Language   | Changelog                                                    |
| ---------- | ------------------------------------------------------------ |
| 🇺🇸 English | Fix bug: Button component does not display loading status when loading property is set to {delay: 0} |
| 🇨🇳 Chinese | 修复bug：Button组件loading属性设置为{ delay: 0 }时不显示加载状态 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at c975345</samp>

Fixed a button loading icon bug and simplified the delay logic in `components/button/button.tsx`.

### <samp>🤖 Generated by Copilot at c975345</samp>

* Fix button loading icon bug when delay prop is positive ([link](https://github.com/ant-design/ant-design/pull/45282/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L78-R82))

### CODE REVIEW

**Comment by afc163** [comment]
来个测试用例

**Comment by YDFlame13** [comment]
> 来个测试用例
https://codesandbox.io/s/dawn-brook-gwmy2q?file=/src/App.js

File: components/button/button.tsx [hunk 1]
```diff
@@ -75,11 +75,11 @@ type LoadingConfigType = {
...
 function getLoadingConfig(loading: BaseButtonProps['loading']): LoadingConfigType {
   if (typeof loading === 'object' && loading) {
-    const delay = loading?.delay;
-    const isDelay = !Number.isNaN(delay) && typeof delay === 'number';
+    let delay = loading?.delay;
+    delay = !Number.isNaN(delay) && typeof delay === 'number' ? delay : 0;
     return {
-      loading: false,
-      delay: isDelay ? delay : 0,
+      loading: !(delay > 0),
```
Comment (kiner-tang) [inline components/button/button.tsx:81]:
```suggestion
      loading: delay <= 0,
```

Comment (YDFlame13) [reply components/button/button.tsx:81]: 😅

**Comment by kiner-tang** [comment]
> > 来个测试用例
> > https://codesandbox.io/s/dawn-brook-gwmy2q?file=/src/App.js

意思是需要验证一下测试用例用来验证本次更改，可以在组件的 __tests__ 目录添加：
image

**Comment by YDFlame13** [comment]
> > > 来个测试用例
> > > https://codesandbox.io/s/dawn-brook-gwmy2q?file=/src/App.js
> 
> 意思是需要验证一下测试用例用来验证本次更改，可以在组件的 **tests** 目录添加： image

好的，谢谢大佬

**Review by afc163 — APPROVED** [review]

**Pull request merged**